store.go

```diff
@@ -2659,6 +2720,50 @@ func init() {
 	if config.Storage.GraphRoot != "" {
 		DefaultStoreOptions.GraphRoot = config.Storage.GraphRoot
 	}
+	if config.Storage.Options.Thinpool.AutoExtendPercent != "" {
+		DefaultStoreOptions.GraphDriverOptions = append(DefaultStoreOptions.GraphDriverOptions, fmt.Sprintf("dm.thinp_autoextend_percent=%s", config.Storage.Options.Thinpool.AutoExtendPercent))
+	}
+
+	if config.Storage.Options.Thinpool.AutoExtendThreshold != "" {
+		DefaultStoreOptions.GraphDriverOptions = append(DefaultStoreOptions.GraphDriverOptions, fmt.Sprintf("dm.thinp_autoextend_threshold=%s", config.Storage.Options.Thinpool.AutoExtendThreshold))
+	}
+
+	if config.Storage.Options.Thinpool.BaseSize != "" {
+		DefaultStoreOptions.GraphDriverOptions = append(DefaultStoreOptions.GraphDriverOptions, fmt.Sprintf("dm.basesize=%s", config.Storage.Options.Thinpool.BaseSize))
+	}
+	if config.Storage.Options.Thinpool.BlockSize != "" {
+		DefaultStoreOptions.GraphDriverOptions = append(DefaultStoreOptions.GraphDriverOptions, fmt.Sprintf("dm.blocksize=%s", config.Storage.Options.Thinpool.BlockSize))
+	}
+	if config.Storage.Options.Thinpool.DirectLvmDevice != "" {
+		DefaultStoreOptions.GraphDriverOptions = append(DefaultStoreOptions.GraphDriverOptions, fmt.Sprintf("dm.directlvm_device=%s", config.Storage.Options.Thinpool.DirectLvmDevice))
+	}
+	if config.Storage.Options.Thinpool.DirectLvmDeviceForce != "" {
+		DefaultStoreOptions.GraphDriverOptions = append(DefaultStoreOptions.GraphDriverOptions, fmt.Sprintf("dm.directlvm_device_force=%s", config.Storage.Options.Thinpool.DirectLvmDeviceForce))
+	}
+	if config.Storage.Options.Thinpool.Fs != "" {
+		DefaultStoreOptions.GraphDriverOptions = append(DefaultStoreOptions.GraphDriverOptions, fmt.Sprintf("dm.fs=%s", config.Storage.Options.Thinpool.Fs))
+	}
+	if config.Storage.Options.Thinpool.LogLevel != "" {
+		DefaultStoreOptions.GraphDriverOptions = append(DefaultStoreOptions.GraphDriverOptions, fmt.Sprintf("dm.fs=libdm_log_level", config.Storage.Options.Thinpool.LogLevel))
+	}
+	if config.Storage.Options.Thinpool.MinFreeSpace != "" {
+		DefaultStoreOptions.GraphDriverOptions = append(DefaultStoreOptions.GraphDriverOptions, fmt.Sprintf("dm.min_free_space=5s", config.Storage.Options.Thinpool.MinFreeSpace))
```

Comment:
This doesn't look right.  It should be a percentage-form value taken from the configuration.

Reply:
Fixed